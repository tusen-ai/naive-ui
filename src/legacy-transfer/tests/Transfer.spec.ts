import { mount } from '@vue/test-utils'
import { sleep } from 'seemly'
import { NLegacyTransfer } from '../index'

describe('n-legacy-transfer', () => {
  it('should work with import on demand', () => {
    mount(NLegacyTransfer)
  })

  it('should work with `disabled` prop', () => {
    const wrapper = mount(NLegacyTransfer, { props: { disabled: true } })
    expect(wrapper.find('.n-legacy-transfer').attributes('class')).toContain(
      'n-legacy-transfer--disabled'
    )
    wrapper.unmount()
  })

  it('should work with `filterable` prop', () => {
    const wrapper = mount(NLegacyTransfer, { props: { filterable: true } })
    expect(wrapper.find('.n-legacy-transfer').attributes('class')).toContain(
      'n-legacy-transfer--filterable'
    )
    wrapper.unmount()
  })

  it('should work with `filter` prop', async () => {
    const options = [
      {
        label: 'test1',
        value: 'test1'
      }
    ]
    const onFilter = jest.fn()
    const wrapper = mount(NLegacyTransfer, {
      props: { filterable: true, filter: onFilter, options }
    })
    await wrapper.find('input').setValue('1')
    await sleep(300)
    expect(onFilter).toHaveBeenCalled()
    wrapper.unmount()
  })

  it('should work with `size` prop', async () => {
    ;(['small', 'medium', 'large'] as const).forEach((i) => {
      const wrapper = mount(NLegacyTransfer, {
        props: { size: i }
      })
      expect(
        wrapper.find('.n-legacy-transfer').attributes('style')
      ).toMatchSnapshot()
      wrapper.unmount()
    })
  })

  it('should work with `source-filter-placeholder`、`target-filter-placeholder` props', async () => {
    const wrapper = mount(NLegacyTransfer, {
      props: {
        filterable: true,
        'source-filter-placeholder': 'test-source',
        'target-filter-placeholder': 'test-target'
      }
    })
    expect(wrapper.findAll('input')[0].attributes('placeholder')).toBe(
      'test-source'
    )
    expect(wrapper.findAll('input')[1].attributes('placeholder')).toBe(
      'test-target'
    )
    wrapper.unmount()
  })

  it('should work with `source-title`、`target-title` props', async () => {
    const wrapper = mount(NLegacyTransfer, {
      props: {
        'source-title': 'test-source',
        'target-title': 'test-target'
      }
    })
    expect(
      wrapper.findAll('.n-legacy-transfer-list-header__header')[0].text()
    ).toBe('test-source')
    expect(
      wrapper.findAll('.n-legacy-transfer-list-header__header')[1].text()
    ).toBe('test-target')
    wrapper.unmount()
  })
})
