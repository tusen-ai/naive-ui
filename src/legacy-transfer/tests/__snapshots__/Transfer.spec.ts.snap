// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`n-legacy-transfer > should work with \`size\` prop 1`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-bezier-ease-in: cubic-bezier(.4, 0, 1, 1); --n-bezier-ease-out: cubic-bezier(0, 0, .2, 1); --n-border-color: rgb(224, 224, 230); --n-border-radius: 3px; --n-extra-font-size: 12px; --n-filter-divider-color: rgb(224, 224, 230); --n-font-size: 14px; --n-header-color: rgba(250, 250, 252, 1); --n-header-extra-text-color: rgb(51, 54, 57); --n-header-font-weight: 400; --n-header-text-color: rgb(31, 34, 37); --n-header-text-color-disabled: rgba(194, 194, 194, 1); --n-item-color-pending: rgb(243, 243, 245); --n-item-height: 28px; --n-item-text-color: rgb(51, 54, 57); --n-item-text-color-disabled: rgba(194, 194, 194, 1); --n-list-color: #fff; --n-width: 440px; --n-icon-color: rgba(194, 194, 194, 1); --n-icon-color-disabled: rgba(209, 209, 209, 1);"`;

exports[`n-legacy-transfer > should work with \`size\` prop 2`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-bezier-ease-in: cubic-bezier(.4, 0, 1, 1); --n-bezier-ease-out: cubic-bezier(0, 0, .2, 1); --n-border-color: rgb(224, 224, 230); --n-border-radius: 3px; --n-extra-font-size: 12px; --n-filter-divider-color: rgb(224, 224, 230); --n-font-size: 14px; --n-header-color: rgba(250, 250, 252, 1); --n-header-extra-text-color: rgb(51, 54, 57); --n-header-font-weight: 400; --n-header-text-color: rgb(31, 34, 37); --n-header-text-color-disabled: rgba(194, 194, 194, 1); --n-item-color-pending: rgb(243, 243, 245); --n-item-height: 34px; --n-item-text-color: rgb(51, 54, 57); --n-item-text-color-disabled: rgba(194, 194, 194, 1); --n-list-color: #fff; --n-width: 440px; --n-icon-color: rgba(194, 194, 194, 1); --n-icon-color-disabled: rgba(209, 209, 209, 1);"`;

exports[`n-legacy-transfer > should work with \`size\` prop 3`] = `"--n-bezier: cubic-bezier(.4, 0, .2, 1); --n-bezier-ease-in: cubic-bezier(.4, 0, 1, 1); --n-bezier-ease-out: cubic-bezier(0, 0, .2, 1); --n-border-color: rgb(224, 224, 230); --n-border-radius: 3px; --n-extra-font-size: 12px; --n-filter-divider-color: rgb(224, 224, 230); --n-font-size: 15px; --n-header-color: rgba(250, 250, 252, 1); --n-header-extra-text-color: rgb(51, 54, 57); --n-header-font-weight: 400; --n-header-text-color: rgb(31, 34, 37); --n-header-text-color-disabled: rgba(194, 194, 194, 1); --n-item-color-pending: rgb(243, 243, 245); --n-item-height: 40px; --n-item-text-color: rgb(51, 54, 57); --n-item-text-color-disabled: rgba(194, 194, 194, 1); --n-list-color: #fff; --n-width: 440px; --n-icon-color: rgba(194, 194, 194, 1); --n-icon-color-disabled: rgba(209, 209, 209, 1);"`;
